### DISCUSSION TITLE
how to begin with windows release pack

### DISCUSSION THREAD

**Discussion by dengzheng-cloud** [ggml-org/llama.cpp#3376] [Q&A]
# Prerequisites

just download release win avx2 pack, and download int4 llama2 7b model via net.
got one question about how to run llama.cpp with model i downloaded.
do i need compile all project like i did in ubuntu, or i just need download visual studio( have no idea about)

# Expected Behavior

is there any way i could just enter ./llama.cpp ./model_path  in cmd then i can begin to chat with model.

# Current Behavior

just extract whole pack file, use this for my uncle who has no computer science context, so better without compile, thx
i only know how to use it in ubuntu, if you know about windows release pack usage, please comment, thank you, sincerely.

**Comment by staviq** [comment]
You extract your download, and open terminal window from there, and from that point, you use it like on Linux.

Fastest way to open a terminal from a folder on windows, is to open the folder, and shift+right-click on empty space between icons, and you'll get "open power shell here" option in the context menu.

There is no `llama.cpp` program, you get couple of difrent example binaries.

Typically, only two of them will be useful for you: `main.exe` or `server.exe`. If you run any of those, like so `main.exe -h` you will get a help text describing all parameters.

Minimal command line invocation would be something like this:

`main.exe -m "patch_to_model" -p "your prompt"`

If you want chat, I recommend using the server program, as it gives you web browser UI.

`server.exe -m "path_to_model"`

**Reply by dengzheng-cloud** [reply]
appreciate it, model downloaded is a little bit older, got magic number error, i'll try another model.
